src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -630,6 +630,27 @@ void dcd_sof_enable(uint8_t rhport, bool en)
 /* DCD Endpoint port
  *------------------------------------------------------------------*/
 
+// Check is IN/OUT endpoint is avaliable before opening it
+static bool dcd_edpt_check_if_avaliable(uint8_t rhport, uint8_t direction)
+{
+  if (direction) {  // IN direction
```

Comment:
Based on the changes, the logic in this function will be: 

```suggestion
static bool dcd_ep_available(uint8_t rhport, uint8_t dir) 
{
    // Verify that we have a vacant EP 
    if ((dwc_ep_config[rhport].in_ep + dwc_ep_config[rhport].out_ep) > (dwc_ep_config[rhport].ep_count_max - 1)) {
        return false;
    }
    // Get the ep_count amount at the moment as a temporal variable
    uint8_t new_ep_count = (dir) ? dwc_ep_config[rhport].in_ep : dwc_ep_config[rhport].out_ep;
    // Increase temporal variable
    new_ep_count++;
    // Verify overflow for IN EP ESP32Sx
 #if TU_CHECK_MCU(OPT_MCU_ESP32S2, OPT_MCU_ESP32S3)
     // ESP32Sx has 6 endpoints, from which only 5 can be confiugred as IN
     if ((dir) && (new_ep_count > (dwc_ep_config[rhport].ep_max_count - 1))) {
         return false;
     }
 #endif //    
    // Write new value back
    if(dir) {
        dwc_ep_config[rhport].in_ep = new_ep_count;
    } else {
        dwc_ep_config[rhport].out_ep = new_cp_count;
    }
    return true; 
}
```